AwaitExpception.Test/Tests.cs

```diff
@@ -34,5 +35,38 @@ public async Task TestFreeFallExceptionThrow()
             // make sure we have waited long enough
             Assert.True(thrower.AwaitCompleted);
         }
+
+        // This test fails the process with exception, not yet found a way to run it normally as a test.
+        // Comment that out if need to run other tests
+        [Test]
+        public async Task TestExceptionThrowInAsyncVoid()
+        {
+            var thrower = new ExceptionThrower();
+            AppDomain.CurrentDomain.UnhandledException += (object sender, UnhandledExceptionEventArgs e) => 
+            {
+                // Unfortunately I haven't yet found the way to not fail here. 
+                // Looks like the exception is caught but the process fails despite of that fact
+                Console.WriteLine("The exception is actually caught globally");
+
+                // Make sure thrower finished async operation
+                Assert.True(thrower.AwaitCompleted, "Exception handled globally");
+            };
+
+            // we throw the exception and not awaiting the method
+            try 
+            {
+                thrower.VoidDelayAndThrow(5);
+            }
+            catch 
+            {
+                Console.WriteLine("The exception is actually caught locally");
+
+                // wait for the test to finish
+                await Task.Delay(10);
```

Comment:
Aren't we going to wait only after the exception is thrown, which is never happening?
And should we really catch the exception? I would want the test to fail if the exception happened, and when it is failing - wrap it with `Assert.ThrowsAsync`.

Reply:
I have reworked to the form you asked for, but the problem here may be a bit wider. ```Assert.Throw``` will not catch anything and actually nothing can catch so this case is a guarantee of crash because it shows an uncontrollable flow of exception in this pattern and actually that was a key part of the test - it just can't be run as a normal test (at least I could not yet find a way how to do so).